ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5399,7 +5399,12 @@ public void cleanUpOneDirectoryForReplace(Path path, FileSystem fs,
     if (isNeedRecycle && conf.getBoolVar(HiveConf.ConfVars.REPLCMENABLED)) {
       recycleDirToCmPath(path, purge);
```

Comment:
Do we want to call `recycleDirToCmPath` for a path that does not exist? More general, do we want to call `deleteOldPathForReplace` when the path does not exist?

Reply:
Thank you for catching this. It looks like `recycleDirToCmPath` might also failed if a path does not exist. The intention was to avoid one additional file system call, however, it seems unavoidable. I added the path check before calling `cleanUpOneDirectoryForReplace`.